### PR TITLE
Update C semantics to use OCAML backend + KORE

### PR DESCRIPTION
**Pull request by dwightguth**
@bmmoore FYI, feel free to take a look at the makefile changes

@andreistefanescu @chathhorn please review

Let me know if you want me to refactor the history into any particular form to make it easier to read.

### CODE REVIEW

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Jenkins: test this please

File: semantics/language/execution/io-buffered.k [hunk 1]
```diff
@@ -102,217 +88,190 @@ module C-IO-BUFFERED
...
                // TODO(chathhorn): do we really need to check this stuff every
                // byte?
-               => assert(notBool (loc(Base, Offset) in Locs),
-                    errorUnseqEffects)
-               ~> assert(Offset <Int Len, errorReadOutOfBounds)
-               ~> assert((notBool (Volatile in getQualsAtOffset(ObjT, Offset)))
-                    orBool isVolatileType(T),
-                    errorNonVolatileLVal)
-               ~> tv(V, t(.Set, no-type))
+               => assertSeq(loc(Base, Offset), Locs)
```
Comment (dwightguth) [inline semantics/language/execution/io-buffered.k]: @andreistefanescu is this what you had in mind? Or do you still have some issue even with this version?

Comment (andreistefanescu) [inline semantics/language/execution/io-buffered.k]: Looks good. Thanks @dwightguth

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by chathhorn** [comment]
This looks good to me!

**Pull request merged**